compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -603,13 +596,11 @@ impl Loop {
 
     /// Count the number of instructions in the loop, including the terminating jumps.
     fn count_all_instructions(&self, function: &Function) -> usize {
-        self.blocks
-            .iter()
-            .map(|block| {
-                let block = &function.dfg[*block];
-                block.instructions().len() + block.terminator().map(|_| 1).unwrap_or_default()
-            })
-            .sum()
+        let iter = self.blocks.iter().map(|block| {
+            let block = &function.dfg[*block];
+            block.instructions().len() + block.terminator().is_some() as usize
+        });
+        iter.sum()
```

Comment:
Is the breaking up of `.iter()` here and with `instructions` above something we should be doing consistently, or just something up to personal preference on a case-by-case basis?

Reply:
It's just to nudge the formatter into not separating `map` and it's brackets onto separate lines just because there's a `sum` afterward. Without it, it gets formatted to:
```rust
        self.blocks
            .iter()
            .map(|block| {
                let block = &function.dfg[*block];
                block.instructions().len() + block.terminator().is_some() as usize
            })
            .sum()
```

Reply:
Indeed, I was just wondering if this is something that I should strive to achieve, or it's not a big deal if it appears on separate lines either.

Reply:
I just think this formatting choice by rustfmt is a bit ugly so I avoid it in my code at least. It's just a personal preference though and I've no actual objective reasons backing me up besides the code being a tad shorter.

Reply:
Okay, just wanted to make sure I'm not systematically causing debt for others to clean up after me 🙂